v2/cmd/nuclei/main.go

```diff
@@ -231,8 +234,28 @@ on extensive configurability, massive extensibility and ease of use.`)
 			gologger.Warning().Msgf("Could not read ignore file from custom path: %s\n", err)
 		}
 	}
+	cleanupOldResumeFiles()
+}
+func cleanupOldResumeFiles() {
```

Comment:
I think we can reuse the logic at https://github.com/projectdiscovery/fileutil/blob/ba188a0c8abc4e8cde3d8b8163e7ea08c7e1cd99/file.go#L44
The only missing part is the prefix check. We could extend the method and make it more generic by adding a new `FileFilters` argument accepting various conditions like:
```go
type FileFilters struct {
    OlderThan time.Duration
    Prefix string
    Suffix string
    RegexPattern string
    CustomCheck func(filename string) bool
    Callback func(filename string) error
}
```
In cascade, if they match, the `Callback` will be invoked (in this case, it will call `os.RemoveAll`). What do you think?

Reply:
I think that's a good idea, please take a look at the new implementation.